### PR TITLE
style-guide: Add chapter about formatting for nightly-only syntax

### PR DESCRIPTION
**Pull request by compiler-errors**
cc @rust-lang/style

nightly policy mentioned below is being proposed in https://github.com/rust-lang/style-team/pull/180

### CODE REVIEW

**Comment by compiler-errors** [comment]
@rfcbot concern all-boxes-should-be-checked

**Review by calebcartwright — APPROVED** [review]

**Comment by joshtriplett** [comment]
:+1: for documenting current practice. I'm not sure how this aligns with us actually documeting nightly style somewhere, though. https://github.com/rust-lang/style-team/pull/180/files says "The style guide may or may not prescribe formatting behavior for nightly syntax".

File: src/doc/style-guide/src/nightly.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+The style guide does not prescribe formatting for nightly-only syntax. Refer to
+the style team policy for nightly formatting procedure regarding breaking
+formatter changes and feature stabilization.
```
Comment (compiler-errors) [inline src/doc/style-guide/src/nightly.md:3]:
```suggestion
Refer to
the style team policy for nightly formatting procedure regarding breaking
formatter changes and feature stabilization.

Style for nightly-only syntax should not be integrated into the style guide until stabilization of the syntax. Instead, style for nightly syntax may be staged below when it has been authored, in a section titled with the feature gate name and tracking issue.
```

Comment (compiler-errors) [reply src/doc/style-guide/src/nightly.md:3]: Thoughts @joshtriplett?

**Comment by joshtriplett** [comment]
@rfcbot concern wait-for-revised-text

@compiler-errors is currently overhauling this text; would like to wait to read the new version.

File: src/doc/style-guide/src/nightly.md (hunk 1 above)
Comment (compiler-errors) [reply src/doc/style-guide/src/nightly.md:3]: Actually, may take a second to integrate the text you shared below.

Comment (compiler-errors) [reply src/doc/style-guide/src/nightly.md:3]: Done.

**Comment by joshtriplett** [comment]
@rfcbot resolved wait-for-revised-text
@rfcbot reviewed

**Comment by compiler-errors** [comment]
@rfcbot resolve all-boxes-should-be-checked

**Comment by compiler-errors** [comment]
I think we can merge this since the FCP is over. Anyone got a spare r+?

@bors rollup

**Pull request merged**